### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte

### ISSUE THREAD

**Issue by wczajus** [mnh-jansson/open-battery-information#2]
Hi, Can not read 'static data', 'battery data' getting the error like below:
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte. (full log attached) 
[error.txt](https://github.com/user-attachments/files/16346834/error.txt)

The UI is populated with values 65.535 and 0.65535 for voltage difference (screen attached) 
![MakitaUI](https://github.com/user-attachments/assets/b57b8b27-17c2-4e37-bb63-85993c53bd6b)

Arduino uno board flashed without any issues log attached ([unoFlash.log](https://github.com/user-attachments/files/16346798/unoFlash.log))

Interface build according schemat attached to the project.

Connected to working battery without any faults BL1380 and BL1840B

Any clue what is worng - I'm guessing it's a communication problem between the interface and the battery because when the interface is disconnected from the battery I get the same error and values ​​in the UI

**Comment by mnh-jansson** [comment]
Thanks for the detailed issue.

Could you verify that the batteries are supported?
https://github.com/mnh-jansson/open-battery-information/wiki/Makita#makita-lxt

I'll try to add support for reading battery information for older batteries ASAP. I don't have the unlock for these yet, but with the battery information it is possible to indicate to the user what is possible.

**Comment by wczajus** [comment]
The battery what I tested is marked as supported on the Wiki (LIPW014 Microcontroller: MAK02).
I am getting the same error when battery is connected and disconnected from interface.

**Comment by mnh-jansson** [comment]
Ok, i think you are right in the guess that the problem is between the battery and the interface. Can you double check the wiring? Please note that the negative battery terminal needs to be connected to GND.

**Comment by wczajus** [comment]
I checked the wiring and made new one from scratch. I managed to connect only once.
Reading information about the battery and controlling the LEDs worked for that session.
Then I hit 'Reset errors' and 'Reset battery message' after that I can't connect again. I get the same error as at the beginning.
"  File "c:\Users\XXX\OneDrive\Dokumenty\GitHub\open-battery-information\OpenBatteryInformation\modules\Makita.py", line 65, in on_read_static_click
    model = response[2:9].decode('utf-8')
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte"

**Comment by mnh-jansson** [comment]
Is the battery still signaling error with double flashing LEDs? Seems like you definitely have some connection issues between the interface and battery.

Reset errors should be sufficient to get the battery to stop signaling fault. Reset battery message might be needed if the charger is rejected the battery even though the battery itself reports fine.

**Comment by wczajus** [comment]
Yes, the battery is still flashing the LEDs.
What should be the connection order or it doesn't matter?
In my case I first connect battery to interface and the power on the Arduino after a while I connect via the UI.
I was also trying to reset Arduino by clicking the reset button on board but it also doesn't help still getting the error and 65 values in UI.

**Comment by mnh-jansson** [comment]
The connection order you describe is the same as i use, but it should not matter.

Do you mind including a picture of the interface?

**Comment by wczajus** [comment]
Sure, the bread bord just created, both resistors are 4,7k (on the picture one is different but now both are the same).
For the board with SMD, battery minus is directly connected to the Arduino.

![IMG_20240724_113627](https://github.com/user-attachments/assets/45a3bb20-1c17-4f99-ab40-c4c62a92121c)
![IMG_20240724_113426](https://github.com/user-attachments/assets/65e345dd-1a14-4a0a-936c-21c4f2dd93c3)
![Messenger_creation_8c566bb9-a37a-48d0-a756-e0ff09ddabdc](https://github.com/user-attachments/assets/c651ea35-6cce-4cac-84c9-f8e71644781f)

**Comment by mnh-jansson** [comment]
Very strange, everything seems to be setup correctly. Maybe reduce the amount of connection by connecting GND directly from the arduino to the battery.

I just noticed that the pull-up on my interface for the Onewire is 10kohm, but i doubt that would make a difference. The only other thing i can think of is some dodgy connection on either the breadboard or the makita connector

**Comment by wczajus** [comment]
You were right.
The problem was with makita connector. The connector itself works fine as I checked resistance between wire and contact but the problem occurred after connecting to the battery the circuit was broken.
I removed the makita connector and made shortcut directly to the battery connector and got it working.
I have tested the 10kohm on the Onewire and works same way as 4.7kohm.
Thanks for your support!

**Comment by mnh-jansson** [comment]
Glad to hear it is working! I'll keep the issue open until i've added correct error handling of connection issues.

**Comment by mnh-jansson** [comment]
@wczajus if you have time, please test the latest version. I'm closing this issue for now.

**Comment by Marvv89** [comment]
I'm having the some UTF-8 errors. At first I tried a BL1830 that looks a bit like the LIPW017, but without LED's. 
Next I tried a working (fully assembled BL1850B) also without success, but since charging is still working fine it seems to be very unlikely that the yellow connector is not soldered well enough anymore.

Checked with a scope, sending the messages is going already, but receiving 0xFF's back.
This evening I will also try the new version software to see if there is any progress.
